app/modules/video-players/jwplayer.js

```diff
@@ -0,0 +1,100 @@
+import Ads from '../ads';
+import BasePlayer from './base';
+import JWPlayerVideoAds from './jwplayer-video-ads';
+import {track} from '../../utils/track';
+
+export const jwPlayerAssets = {
+	styles: '/mobile-wiki/assets/jwplayer/index.css',
+	script: '/mobile-wiki/assets/jwplayer/wikiajwplayer.js'
+};
+
+export default class JWPlayer extends BasePlayer {
+	constructor(provider, params) {
+		const originalOnCreate = params.onCreate;
+
+		super(provider, params);
+
+		params.onCreate = (bidParams, player) => {
+			const adsInstance = Ads.getInstance();
+
+			originalOnCreate(player);
+
+			adsInstance.jwPlayerAds(player, bidParams);
+			adsInstance.jwPlayerMoat(player);
+		};
+
+		this.adTrackingParams = params.adTrackingParams || {};
+	}
+
+	setupPlayer() {
+		if (!window.wikiaJWPlayer) {
+			this.loadPlayer();
+		} else {
+			this.createPlayer();
+		}
+	}
+
+	/**
+	 * @returns {void}
+	 */
+	createPlayer() {
+		Ads.getInstance()
+			.waitForReady()
+			.then(() => (new JWPlayerVideoAds(this.params)).getConfig())
+			.then(this.initializePlayer.bind(this));
+	}
+
+	initializePlayer(bidParams) {
+		window.wikiaJWPlayer(this.params.containerId, {
+				tracking: {
```

Comment:
How about extracting the config here to a separate builder function? It could return a configuration object and this could make setup a bit more readable.

Reply:
I don't see much value in separate func in that case - @latata, @hakubo?